rpcserver.go

```diff
@@ -2711,34 +2711,60 @@ func (r *rpcServer) WalletBalance(ctx context.Context,
 // ChannelBalance returns the total available channel flow across all open
 // channels in satoshis.
 func (r *rpcServer) ChannelBalance(ctx context.Context,
-	in *lnrpc.ChannelBalanceRequest) (*lnrpc.ChannelBalanceResponse, error) {
+	in *lnrpc.ChannelBalanceRequest) (
+	*lnrpc.ChannelBalanceResponse, error) {
+
+	var (
+		localBalance             btcutil.Amount
+		remoteBalance            btcutil.Amount
+		unsettledBalance         btcutil.Amount
+		pendingOpenLocalBalance  btcutil.Amount
+		pendingOpenRemoteBalance btcutil.Amount
+	)
 
 	openChannels, err := r.server.chanDB.FetchAllOpenChannels()
 	if err != nil {
 		return nil, err
 	}
 
-	var balance btcutil.Amount
 	for _, channel := range openChannels {
-		balance += channel.LocalCommitment.LocalBalance.ToSatoshis()
+		c := channel.LocalCommitment
+		localBalance += c.LocalBalance.ToSatoshis()
+		remoteBalance += c.RemoteBalance.ToSatoshis()
+
+		// Add pending htlc amount.
+		for _, htlc := range c.Htlcs {
+			unsettledBalance += htlc.Amt.ToSatoshis()
```

Comment:
do we want to split this into unsettled_local and unsettled_remote? we can separate them based on the `Incoming` bool on the htlc, e.g. an `Incoming` htlc would count towards unsettled_local